sw/device/lib/crypto/impl/mlkem/fips202_glue.h

```diff
@@ -0,0 +1,77 @@
+// Copyright The mlkem-native project authors
+// Copyright zeroRISC Inc.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef MLK_FIPS202_FIPS202_H
+#define MLK_FIPS202_FIPS202_H
+#include <stdint.h>
+
+#include "mlkem/src/common.h"
+#include "sw/device/lib/crypto/drivers/kmac.h"
+#include "sw/device/lib/crypto/include/sha3.h"
+
+#define SHAKE128_RATE 168
+
+typedef struct {
+} MLK_ALIGN mlk_shake128ctx;
+
+static MLK_INLINE void mlk_shake128_absorb_once(mlk_shake128ctx *state,
+                                                const uint8_t *input,
+                                                size_t inlen) {
+  (void)state;
+  kmac_shake128_start();
+  kmac_absorb(input, inlen);
+  kmac_process();
+}
+
+static MLK_INLINE void mlk_shake128_squeezeblocks(uint8_t *output,
+                                                  size_t nblocks,
+                                                  mlk_shake128ctx *state) {
+  (void)state;
+  kmac_squeeze_blocks(nblocks, kHardenedBoolFalse, (uint32_t *)output, NULL);
+}
+
+static MLK_INLINE void mlk_shake128_init(mlk_shake128ctx *state) {
+  (void)state;
+}
+
+static MLK_INLINE void mlk_shake128_release(mlk_shake128ctx *state) {
+  (void)state;
+  kmac_squeeze_end(0, kHardenedBoolFalse, NULL, NULL);
+}
+
+static MLK_INLINE void mlk_shake256(uint8_t *output, size_t outlen,
+                                    const uint8_t *input, size_t inlen) {
+  otcrypto_hash_digest_t md = {.data = (uint32_t *)output,
+                               .len = outlen / 4,
```

Comment:
It would be worth noting, and perhaps extending the CBMC contracts in mlkem-native accordingly, that `outlen` will be divisible by 4 for all call-sites from mlkem-native.